Open Judge System/Workers/OJS.Workers.LocalWorker/SubmissionJobStrategy.cs

```diff
@@ -0,0 +1,293 @@
+﻿namespace OJS.Workers.LocalWorker
+{
+    using System;
+    using System.Linq;
+
+    using OJS.Data.Models;
+    using OJS.Services.Data.Participants;
+    using OJS.Services.Data.ParticipantScores;
+    using OJS.Services.Data.Submissions;
+    using OJS.Services.Data.SubmissionsForProcessing;
+    using OJS.Services.Data.TestRuns;
+    using OJS.Workers.ExecutionStrategies;
+    using OJS.Workers.Jobs;
+    using OJS.Workers.Jobs.Models;
+
+    public class SubmissionJobStrategy : BaseJobStrategy<int>
+    {
+        private readonly ISubmissionsDataService submissionsData;
+        private readonly ITestRunsDataService testRunsData;
+        private readonly IParticipantsDataService participantsData;
+        private readonly IParticipantScoresDataService participantScoresData;
+        private readonly ISubmissionsForProcessingDataService submissionsForProccessingData;
+
+        private Submission submission;
+        private SubmissionForProcessing submissionForProcessing;
+
+        public SubmissionJobStrategy(
+            ISubmissionsDataService submissionsData,
+            ITestRunsDataService testRunsData,
+            IParticipantsDataService participantsData,
+            IParticipantScoresDataService participantScoresData,
+            ISubmissionsForProcessingDataService submissionsForProccessingData)
+        {
+            this.submissionsData = submissionsData;
+            this.testRunsData = testRunsData;
+            this.participantsData = participantsData;
+            this.participantScoresData = participantScoresData;
+            this.submissionsForProccessingData = submissionsForProccessingData;
+        }
+
+        public override SubmissionModel RetrieveSubmission()
+        {
+            bool retrievedSubmissionSuccessfully;
+
+            lock (this.SubmissionsForProcessing)
+            {
+                if (this.SubmissionsForProcessing.IsEmpty)
+                {
+                    var submissions = this.submissionsForProccessingData
+                        .GetAllUnprocessed()
+                        .OrderBy(x => x.Id)
+                        .Select(x => x.SubmissionId)
+                        .ToList();
+
+                    submissions.ForEach(this.SubmissionsForProcessing.Enqueue);
```

Comment:
This can be appended above:
```
this.submissionsForProccessingData
                        .GetAllUnprocessed()
                        .OrderBy(x => x.Id)
                        .Select(x => x.SubmissionId)
                        .ToList()
                        .ForEach(this.SubmissionsForProcessing.Enqueue);
```